bakery/checker_test.go

```diff
@@ -916,12 +912,12 @@ func (c *client) requestMacaroons(svc *service) []macaroon.Slice {
 	return ms
 }
 
-func (c *client) dischargeAll(ctxt context.Context, m *macaroon.Macaroon, ns *checkers.Namespace) (macaroon.Slice, error) {
-	return bakery.DischargeAll(ctxt, m, func(_ context.Context, cav macaroon.Caveat) (*macaroon.Macaroon, error) {
+func (c *client) dischargeAll(ctxt context.Context, m *bakery.Macaroon) (macaroon.Slice, error) {
+	return bakery.DischargeAll(ctxt, m, func(_ context.Context, cav macaroon.Caveat, payload []byte) (*bakery.Macaroon, error) {
 		d := c.dischargers[cav.Location]
 		if d == nil {
 			return nil, errgo.Newf("third party discharger %q not found", cav.Location)
 		}
-		return d.discharge(ctxt, cav, ns)
+		return d.discharge(ctxt, cav, payload)
```

Comment:
not that it matters a great deal, but is there any reason to use the context from the closure rather than the one that's passed to the function.